KomaMRIBase/src/datatypes/Sequence.jl

```diff
@@ -487,12 +491,18 @@ function get_Mk(seq::Sequence, k; Δt=1, skip_rf=zeros(Bool, sum(is_RF_on.(seq))
 		for (rf, p) in enumerate(parts)
 			mk[p,i] = cumtrapz(Δt[p]', [t[p]' t[p[end]]'.+Δt[p[end]]].^k .* G[i][p[1]:p[end]+1]')[:] #This is the exact integral
 			if rf > 1 # First part does not have RF
-				#if !skip_rf[rf-1]
-				if rf_type[rf-1] == 0 # Excitation
+				if rf_type[rf-1] != Undefined()
+					if rf_type[rf-1] == Excitation() # Explicit Excitation
+						mk[p,i] .-= 0
+					elseif rf_type[rf-1] == Refocusing() # Explicit Refocuse
+						mk[p,i] .-= mkf		
+					else
+						mk[p,i] .+= mkf
+					end
+				elseif rf_type[rf-1] == 0 # Excitation
 					mk[p,i] .-= 0
 				elseif rf_type[rf-1] == 1 # Refocuse
 					mk[p,i] .-= mkf
-				#end
 				else
```

Comment:
You have the logic repeated twice, this doesn't make sense, please only use an `rf_type::Vector{RFUse}` (like explained in previous comment).